### PR TITLE
Reduce execution time of MAST remote test suite

### LINKED ISSUE #2209: Fix MAST download products tests to limit test download to a minimum amount
As the comment says in the code the current test is now ideal as it both takes too long and downloads way too much data.
Ideally, the fix is just to choose another OBSID that corresponds to a very small set of file(s). The test has been marked to be skipped for the time being.

https://github.com/astropy/astroquery/pull/2174#discussion_r748654667

cc @jaymedina

### PR DESCRIPTION
**Pull request by snbianco**
Closes #2209 

@bsipocz @astrojimig

Although the issue only mentions one particular test, I ended up making a more changes to the entire suite to reduce running time. I got it down to around 6 minutes on my machine. The bottlenecks are still in the tests that download files, but I think this is a large improvement.

I will look for (or make) a separate issue about the documentation tests. How urgent would this be?

### CODE REVIEW

**Comment by bsipocz** [comment]
> I will look for (or make) a separate issue about the documentation tests. How urgent would this be?

There are a couple of old ones, like e.g. https://github.com/astropy/astroquery/issues/2505, but also please open new ones for specific issues or in any way that you would find useful. 
Nothing is really urgent, as you see some of the issues are open for year now. I just got super excited to see renewed activity and jumped on a few issues that would makes the generic maintainer life easier (e.g. this one).

**Comment by pllim** [comment]
Added. Hope it works! 🤞

**Review by bsipocz — APPROVED** [review]
Thank you so much, this is a great improvement.

However, locally I run into timeouts for the last 5 tests. However, I see the same timeouts on the current `main`, so would go ahead merging this PR as is as leave digging into those timeouts for a follow-up.

```
=============================================== short test summary info ================================================
FAILED astroquery/mast/tests/test_mast_remote.py::TestMast::test_zcut_get_surveys - requests.exceptions.HTTPError: 504 Server Error: Gateway Time-out for url: https://mast.stsci.edu/zcut/api/v0.1/survey
FAILED astroquery/mast/tests/test_mast_remote.py::TestMast::test_zcut_download_cutouts - requests.exceptions.HTTPError: 504 Server Error: Gateway Time-out for url: https://mast.stsci.edu/zcut/api/v0.1/ast...
FAILED astroquery/mast/tests/test_mast_remote.py::TestMast::test_zcut_get_cutouts - requests.exceptions.HTTPError: 504 Server Error: Gateway Time-out for url: https://mast.stsci.edu/zcut/api/v0.1/ast...
FAILED astroquery/mast/tests/test_mast_remote.py::TestMast::test_hapcut_download_cutouts - requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://mast.stsci.edu/hapcut/api/v...
FAILED astroquery/mast/tests/test_mast_remote.py::TestMast::test_hapcut_get_cutouts - requests.exceptions.HTTPError: 500 Server Error: Internal Server Error for url: https://mast.stsci.edu/hapcut/api/v...
================================ 5 failed, 96 passed, 42 warnings in 1328.38s (0:22:08) ================================
```

File: CHANGES.rst [hunk 1]
```diff
@@ -87,6 +87,7 @@ mast
 ^^^^
 
 - Fix bug in which the ``local_path`` parameter for the ``mast.observations.download_file`` method does not accept a directory. [#3016]
+- Optimize remote test suite to improve performance and reduce execution time. [#3036]
```
[the remark this answers is not in the document]
Comment (bsipocz) [reply CHANGES.rst:90]: I'll remove this at release time.

**Pull request merged**